### PR TITLE
Add tool tips to sidebar tree items

### PR DESCRIPTION
**Pull request by uklotzde**
A byproduct from the aoide branch, simple and straightforward.

Could be added even though we will migrate to QML.

### CODE REVIEW

**Comment by Be-ing** [comment]
I'm in doubt we should merge this unless you're planning to add those tooltips. Adding code that isn't used doesn't help anything.

**Comment by uklotzde** [comment]
> I'm in doubt we should merge this unless you're planning to add those tooltips. Adding code that isn't used doesn't help anything.

The current state is incomplete and somwhow inconsistent. Displaying the data for the Qt::ToolTipRole might be surprising.

I would prefer a consistent and feature complete state, even if it is not used.

**Comment by ronso0** [comment]
Thanks for starting this. I can imagine to work on creating the actual tooltips.
Currently those are playlist/crate Ids, not very helpful (except for debugging purposes like for #4193). Showing `playlist name #tracks, duration` exactly like in the sidebar may seem redundant but it can be helpful if the sidebar is too narrow to show that info

File: src/library/sidebarmodel.cpp [hunk 1]
```diff
@@ -257,8 +257,13 @@ QVariant SidebarModel::data(const QModelIndex& index, int role) const {
             // If it's the "Quick Links" node, display it's name
             if (pTreeItem->getData().toString() == QUICK_LINK_NODE) {
                 return pTreeItem->getLabel();
+            } else {
+                if (pTreeItem->getToolTip().isEmpty()) {
+                    return pTreeItem->getData();
+                } else {
+                    return pTreeItem->getToolTip();
+                }
```
Comment (ronso0) [inline src/library/sidebarmodel.cpp:265]: Let's hide the useless Ids, except when debugging
```suggestion
            } else {
                if (!pTreeItem->getToolTip().isEmpty()) {
                    return pTreeItem->getToolTip();
                } else if (CmdlineArgs::Instance().getDeveloper()) {
                    return pTreeItem->getData();
                }
```

**Review by ronso0 — CHANGES_REQUESTED** [review]

File: src/library/sidebarmodel.cpp (hunk 1 above)
Comment (uklotzde) [reply src/library/sidebarmodel.cpp]: 775e7c7ff86cc3811238426901e1082043a2f6b9

File: src/library/sidebarmodel.cpp [hunk 2]
```diff
@@ -257,8 +258,17 @@ QVariant SidebarModel::data(const QModelIndex& index, int role) const {
...
+            } else {
+                if (!pTreeItem->getToolTip().isEmpty()) {
+                    return pTreeItem->getToolTip();
+                } else if (CmdlineArgs::Instance().getDeveloper()) {
+                    // Display the internal data only for debugging
+                    return pTreeItem->getData();
+                } else {
+                    // No tool tip
+                    return {};
+                }
```
Comment (ronso0) [inline src/library/sidebarmodel.cpp:270]: @uklotzde Sorry I'm looking into treeitem code just now. I would actually store the label as tooltip, but for that purpose we could simply return the label here (like for the QUICK_LINK_NODE) -- which would make the set/getTooltip methods obsolete...
What exactly do you have in mind with the tooltips?

Comment (uklotzde) [reply src/library/sidebarmodel.cpp:270]: Each aoide entity (like a *playlist* or *collection*) has both a *label* and *notes* (= more verbose description). The *notes* are displayed as a tool tip.

Comment (uklotzde) [reply src/library/sidebarmodel.cpp:270]: This also applies to the *prepared queries* that are stored locally.

**Comment by uklotzde** [comment]
Might not be needed if storing the whole item as data.

**Pull request closed without merging**